File: mobile/ios/ShiftPlanner/ShiftPlanner/Features/Auth/Repositories/AuthRepository.swift
import Foundation

/// Result of a sign-up: either an immediate session, or a pending email
/// confirmation the user must complete before they can log in.
enum SignUpOutcome {
    case loggedIn(AppUser)
    case verificationRequired(email: String, message: String?)
}

protocol AuthRepository {
    func login(email: String, password: String) async throws -> AppUser

    func signUp(email: String, password: String, name: String, role: UserRole) async throws -> SignUpOutcome
    func resendVerification(email: String) async throws

    func logout() async
    func getCurrentUser() async -> AppUser?

    /// Permanently deletes the current account (DELETE auth/me).
    func deleteAccount() async throws
}

File: mobile/android/app/src/main/java/com/froggyriia/shiftplanner/data/auth/AuthRepository.kt
package com.froggyriia.shiftplanner.data.auth

import com.froggyriia.shiftplanner.domain.model.AppUser
import com.froggyriia.shiftplanner.domain.model.UserRole

interface AuthRepository {
    suspend fun login(
        email: String,
        password: String
    ): AppUser

    // Returns null if email verification is required (user must check inbox before logging in)
    suspend fun signUp(
        email: String,
        password: String,
        name: String,
        role: UserRole
    ): AppUser?

    suspend fun logout()

    suspend fun getCurrentUser(): AppUser?

    suspend fun deleteAccount()
}